crates/common/tedge_config/tests/test_tedge_config.rs

```diff
@@ -278,6 +285,7 @@ bind_address = "0.0.0.0"
             config.query(MqttBindAddressSetting)?,
             updated_mqtt_bind_address
         );
+        assert_eq!(config.query(ServiceTypeSetting)?, "systemd");
```

Comment:
```suggestion
        assert_eq!(config.query(ServiceTypeSetting)?, updated_service_type);
```
Together with the change of line 157, removing `.to_string()` and make the type of `updated_service_type` to `&str`.
```
    let updated_service_type = "systemd".to_string();
```

Reply:
fixed